pkg/local_object_storage/blobstor/fstree/fstree_write_linux.go

```diff
@@ -29,19 +58,159 @@ func newSpecificWriteData(root string, perm fs.FileMode, noSync bool) func(strin
 	}
 	_ = unix.Close(fd) // Don't care about error.
 	w := &linuxWriter{
-		root:  root,
-		perm:  uint32(perm),
-		flags: flags,
+		root:   root,
+		perm:   uint32(perm),
+		flags:  flags,
+		bFlags: bFlags,
+		noSync: noSync,
+	}
+	return w
+}
+
+func (w *linuxWriter) newSyncBatch() (*syncBatch, error) {
+	fd, err := unix.Open(w.root, w.bFlags, w.perm)
+	if err != nil {
+		return nil, err
+	}
+	sb := &syncBatch{
+		fd:       fd,
+		procname: "/proc/self/fd/" + strconv.FormatUint(uint64(fd), 10),
+		ready:    make(chan struct{}),
+		noSync:   w.noSync,
+	}
+	sb.lock.Lock()
+	sb.timer = time.AfterFunc(defaultTick, sb.sync)
+	return sb, nil
+}
+
+func (b *syncBatch) sync() {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	select {
+	case <-b.ready:
+		return
+	default:
 	}
-	return w.writeData
+	b.intSync()
 }
 
-func (w *linuxWriter) writeData(p string, data []byte) error {
-	err := w.writeFile(p, data)
-	if errors.Is(err, unix.ENOSPC) {
-		return common.ErrNoSpace
+func (b *syncBatch) intSync() {
+	var err error
+
+	if b.err == nil && !b.noSync {
+		err = unix.Fdatasync(b.fd)
+		if err != nil {
+			b.err = err
+		}
+	}
+
+	err = unix.Close(b.fd)
+	if b.err == nil && err != nil {
+		b.err = err
+	}
+	close(b.ready)
+	_ = b.timer.Stop() // True is stopped, but false is "AfterFunc already running".
+}
+
+func (b *syncBatch) wait() error {
+	<-b.ready
+	return b.err
+}
+
+func (b *syncBatch) write(id oid.ID, p string, data []byte) error {
+	var (
+		err  error
+		pref [1 + len(id) + 4]byte
```

Comment:
we have consts for this numbers but they are defined in a func for some reason

Reply:
It's harder to handle in the reader, more code with more fancy calculations. I had a thought about moving them into upper-layer definitions, but they'd need some `combined` prefix that would make them _so_ long. But can be done.